### PR TITLE
Migrate brave-ios into brave-core

### PR DESCRIPTION
**Pull request by kylehickinson**
Resolves 

Sec review: https://github.com/brave/reviews/issues/1492

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_presubmit_config.json5
Comment (goodov) [inline chromium_presubmit_config.json5:165]: pls move all these into `default_files_to_skip`.

File: build/commands/lib/config.js
Comment (goodov) [inline build/commands/lib/config.js:198]: pls use consistent name everywhere.

File: chromium_presubmit_config.json5
Comment (kylehickinson) [reply chromium_presubmit_config.json5:165]: @goodov I want to eventually add an iOS app specific presubmit in the future, likely to run swiftlint, would adding these to `default_files_to_skip` cause them to be skipped there too or does this config file only affect Chromium specific presubmit checks?

**Comment by kylehickinson** [comment]
`Jenkinsfile` change to be reverted once CI parts merge

**Comment by kylehickinson** [comment]
Also `swiftformat` related changes likely to be reverted after merging so only new files are formatted/checked for diffs on presubmit

File: script/brave_license_helper.py
Comment (fmarier) [inline script/brave_license_helper.py:36]: I think we need to do something like we did for Android deps here (see the `# Add all Android libraries since they're not directly contained within a third_party directory.` block below). That way the licensing script will pick up any new third-party that was added without licensing info.

It doesn't have to block this PR necessarily, but at the very least, we should have an issue filed and a `FIXME` here.

File: script/brave_license_helper.py
Comment (fmarier) [inline script/brave_license_helper.py:152]: These look like they are "vendored in" as opposed to pulled them from external repos in the build step, no?

If so, then we don't need to add them here and instead we can add ` README.chromium` and `LICENSE` files in `third_party/ios_deps/{GRDWireGuardKit,JitsiMeet,Static}`. In fact `Static/LICENSE` already exists.

**Review by fmarier — CHANGES_REQUESTED** [review]

File: script/brave_license_helper.py
Comment (fmarier) [reply script/brave_license_helper.py:152]: Here's an example from the Android side: https://github.com/brave/brave-core/tree/master/third_party/android_deps/libs/com_github_bumptech_glide

Comment (kylehickinson) [reply script/brave_license_helper.py]: Fixed in https://github.com/brave/brave-core/pull/21553/commits/d05f0c407975d4c34d611641a675edf026730a4d

File: script/brave_license_helper.py
Comment (kylehickinson) [reply script/brave_license_helper.py:36]: Fixed in https://github.com/brave/brave-core/pull/21553/commits/d05f0c407975d4c34d611641a675edf026730a4d

One thing I had to do was unfortunately not walk recursively, because unfortunately Jitsi includes a `node_modules` folder inside of their frameworks and it gets wonky… I can swap it up to walk recursively the same way as Android but we'd need to then exclude their node_modules directory for each framework in the xcframework. Instead I made a note that we'd never nest deps for now

**Review by fmarier — APPROVED** [review]
`script/brave_license_helper.py` changes look good.

File: ios/brave-ios/App/Client.xcodeproj/xcshareddata/xcschemes/Nightly.xcscheme
Comment (iccub) [inline ios/brave-ios/App/Client.xcodeproj/xcshareddata/xcschemes/Nightly.xcscheme:13]: I assume `&quot` characters are expected to be here

File: ios/brave-ios/App/l10n/tools/xliff-cleanup.py
Comment (iccub) [inline ios/brave-ios/App/l10n/tools/xliff-cleanup.py:3]: 2024

File: ios/brave-ios/App/l10n/tools/xliff-cleanup.py
Comment (iccub) [inline ios/brave-ios/App/l10n/tools/xliff-cleanup.py:58]: are those kind of changes due to brave-core formatter?

File: ios/brave-ios/Sources/Brave/Frontend/Reader/Reader.css
Comment (iccub) [inline ios/brave-ios/Sources/Brave/Frontend/Reader/Reader.css:1]: i see those outdated licenses in few places, was it automated?
I will skip adding comments about them from now on

File: ios/brave-ios/Sources/Data/models/SessionWindow.swift
Comment (iccub) [inline ios/brave-ios/Sources/Data/models/SessionWindow.swift:68]: is this change supposed to be part of this PR?

File: ios/brave-ios/fastlane/Fastfile
Comment (iccub) [inline ios/brave-ios/fastlane/Fastfile:121]: So we upload to testers automatically always with this testers notes? not for now but we can improve on it, maybe fetch some commit names from git

File: ios/brave-ios/fastlane/Fastfile
Comment (iccub) [inline ios/brave-ios/fastlane/Fastfile:26]: nit: this could be a constant. easier to replace in the future and i see it in few places

File: ios/brave-ios/scripts/bootstrap.sh
Comment (iccub) [inline ios/brave-ios/scripts/bootstrap.sh:14]: unrelated: I wonder if we should have `npm audit` somewhere to remind us of package vulns

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (iccub) [inline ios/brave-ios/scripts/scheme_preaction.py:7]: offtopic, any reason you chose python over swift to write preactions?

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (iccub) [inline ios/brave-ios/scripts/scheme_preaction.py:118]: are we handling those todos now or later? if later we can add tickets for them

File: ios/brave-ios/swiftlint.sh
Comment (iccub) [inline ios/brave-ios/swiftlint.sh:9]: Is swiftlint preserved or we stop using it?

**Review by iccub — COMMENTED** [review]
small comments and questions only.
I ran it branch and everything seems to work well for me

File: Jenkinsfile
Comment (kylehickinson) [reply Jenkinsfile:85]: This will be reverted once CI changes are merged, yep!

File: ios/brave-ios/App/Client.xcodeproj/xcshareddata/xcschemes/Nightly.xcscheme
Comment (kylehickinson) [reply ios/brave-ios/App/Client.xcodeproj/xcshareddata/xcschemes/Nightly.xcscheme:13]: Yes, Xcode scheme editor is writing this file

File: ios/brave-ios/App/l10n/tools/xliff-cleanup.py
Comment (kylehickinson) [reply ios/brave-ios/App/l10n/tools/xliff-cleanup.py:3]: This file was created in 2018 according to git, so copyright is 2018

File: ios/brave-ios/App/l10n/tools/xliff-cleanup.py
Comment (kylehickinson) [reply ios/brave-ios/App/l10n/tools/xliff-cleanup.py:58]: Yep, `npm run format` applies python formatting

File: ios/brave-ios/Sources/Data/models/SessionWindow.swift
Comment (kylehickinson) [reply ios/brave-ios/Sources/Data/models/SessionWindow.swift:68]: There are a number of iOS specific changes in this PR when I was fixing unit tests, so yeah

File: ios/brave-ios/fastlane/Fastfile
Comment (kylehickinson) [reply ios/brave-ios/fastlane/Fastfile:121]: Yes for now I chose this, but we can find a way to pass in info from Jenkins if possible. They already technically have a changelog there, but maybe we can somehow generate a changelog based on files changed in the `ios/` directory instead

File: ios/brave-ios/scripts/bootstrap.sh
Comment (kylehickinson) [reply ios/brave-ios/scripts/bootstrap.sh:14]: I believe that will run automatically when you run `npm run sync ...` commands

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (kylehickinson) [reply ios/brave-ios/scripts/scheme_preaction.py:7]: I actually didnt even think about that surprisingly, I choose python as thats what Chromium prefers when running scripts

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (kylehickinson) [reply ios/brave-ios/scripts/scheme_preaction.py:118]: Later, these todos are actually old and existed in the old bootstrap as well

File: ios/brave-ios/swiftlint.sh
Comment (kylehickinson) [reply ios/brave-ios/swiftlint.sh:9]: For now we will remove it, Chromium iOS only relies on `swiftformat` for their Swift code which I will be enabling after this migration completes (so that it doesnt run on every Swift file we have now).

I may bring back swiftlint in the future and add it as a custom PRESUBMIT check though. It requires pulling the tool through DEPS/runhooks though so I left it for now

**Review by iccub — APPROVED** [review]

File: ios/brave-ios/README.md
Comment (goodov) [inline ios/brave-ios/README.md:9]: I think these two steps should be wrapped in some `npm run ios_open_project` command (feel free to change the name) that will bootstrap and open `xcodeproj` automatically.

everything from `ios/brave-ios/scripts/bootstrap.sh` can be implemented in the npm command. This also means  `npm run ios_pack_js` can be replaced with a direct `webpack` module call from the script.

File: build/commands/lib/util.js
Comment (goodov) [inline build/commands/lib/util.js:885]: please move this into a separate file into `brave/build/commands/scripts` dir (`ios.js` or `iosCommands.js` for ex.).

Comment (kylehickinson) [reply build/commands/lib/util.js]: Fixed here: https://github.com/brave/brave-core/pull/21553/commits/75eff03e08eac9ca46a5b2deed1445359c57a58c

File: ios/brave-ios/scripts/xcode_scheme_preaction.sh
Comment (kylehickinson) [reply ios/brave-ios/scripts/xcode_scheme_preaction.sh:28]: Fixed here: https://github.com/brave/brave-core/pull/21553/commits/c30d30d5d06854dde71c9b472a3f7a74f51c904b I also attempt to fixup the PATH for users who install node/npm via brew on arm64 machines (since brew prefix on those machines are `/opt/homebrew`)

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:64]: do you really need this file for the initial build?

please add explanation here for each thing we're copying. why are we doing it and what exact issue this solves.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:86]: just `main()` and remove all `sys.exit()` calls.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:47]: pls remove this, there's no point in this warning.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:62]: 1. this bootstrapper should support `--force` argument that will rewrite the symlink/directory entirely.
2. it should work correctly if the symlink is broken (re-create everything automatically).
3. it should do nothing if the symlink is valid or the directory already exists and `--force` is not passed.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:60]: will it work? shouldn't this be `['--', '--verbose']`?

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:15]: please remove `SRC_DIR` and replace all `os.path.join` things with `brave_chromium_utils.wspath()` calls.
https://github.com/brave/brave-core/blob/6fd9b197fadf864762ee98211b62a631aab4919d/script/brave_chromium_utils.py#L33-L38

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:56]: I don't think you need tihs.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:57]: this also looks unnecessary.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:73]: contents = inspect.cleandoc(f"""
    # This file is generated by ios_bootstrap.py
    #
    # This allows proper source mapping to builds made with the GN arg
    # strip_absolute_paths_from_debug_symbols set to true.
    settings set target.source-map "../.." "{wspath("//")}"
    """)

File: build/commands/scripts/iosCommands.js
Comment (goodov) [inline build/commands/scripts/iosCommands.js:56]: also add `--force` here.

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (goodov) [inline ios/brave-ios/scripts/scheme_preaction.py:47]: actually I think this call should be the last here, i.e. after `CallNpm(['npm', 'run', 'ios_pack_js'])`.

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (goodov) [inline ios/brave-ios/scripts/scheme_preaction.py:50]: all these commands should be converted into GN actions.

this script should only call `BuildCore` and `UpdateSymlink`.

File: script/ios_bootstrap.py
Comment (kylehickinson) [reply script/ios_bootstrap.py:60]: `--verbose` here is on the npm call specifically, it isn't an argument of `ios_pack_js`, so not `--` needed

File: script/ios_bootstrap.py
Comment (kylehickinson) [reply script/ios_bootstrap.py:64]: Xcode fails stating it can't find the xcconfig (even after preactions create it…), having the file around earlier seems to remedy that

File: script/ios_bootstrap.py
Comment (kylehickinson) [reply script/ios_bootstrap.py:60]: Removed this since it didnt seem useful

File: ios/brave-ios/scripts/scheme_preaction.py
Comment (kylehickinson) [reply ios/brave-ios/scripts/scheme_preaction.py:50]: Opened follow up issue here for this https://github.com/brave/brave-browser/issues/35865

File: script/ios_bootstrap.py
Comment (kylehickinson) [reply script/ios_bootstrap.py]: Added these and tested all these scenarios: https://github.com/brave/brave-core/pull/21553/commits/88060123fd0dcf711dd0b1c78c167a8b7c0b2bdd

File: script/ios_bootstrap.py
Comment (kylehickinson) [reply script/ios_bootstrap.py]: My PR is not fully rebased on master yet so I dont have `brave_chromium_utils` but I did use `import_util` for the time being, will replace with `brave_chromium_utils` on rebase

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:84]: `cleandoc` is here to remove all leading spaces on each line so this can be formatted as a normal text (you need to do it manually):
```
    contents = inspect.cleandoc(f"""
    # This file is generated by ios_bootstrap.py
    #
    # This allows proper source mapping to builds made with the GN arg
    # strip_absolute_paths_from_debug_symbols set to true.
    settings set target.source-map "../.." "{wspath("//")}"
    """)

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:74]: touch only if the file doesn't exist.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:87]: write only if the file doesn't exist or `--force` is passed.

File: script/ios_bootstrap.py
Comment (goodov) [inline script/ios_bootstrap.py:69]: pls make a helper to do `copy_tree()` if a target doesn't exist. Run it explicitly for both `BraveCore.xcframework` and `MaterialComponents.xcframework`.

**Review by goodov — APPROVED** [review]

**Review by mihaiplesa — APPROVED** [review]

File: Jenkinsfile
Comment (mihaiplesa) [reply Jenkinsfile]: Reverted.

**Pull request merged**
